_posts/2020-06-30-Making-IrishGen-SPARQL-Part-II.md

```diff
@@ -0,0 +1,252 @@
+in this next instalment where the `construct` query form will be
+introduced and discussed.
+
+Simply but opaquely stated, the `construct` query form takes one
+sub-graph and returns a new sub-graph which is transformed from the
+original.  This means that you can search the dataset for a particular
+pattern then transform it into another pattern that the user supplies.
+While this might not sound very useful for IrishGen and, by extension,
+Báeth, what it allows the user to do is to extract interesting and
+useful parts of the overall dataset such as Báeth's children or, even
+more interestingly, Báeth's genealogy as reconstructed by the
+reasoner.
+
+We will start by examining Báeth's children so that we can show their
+relationship to Baeth.  In this, we will need to choose a particular
+Báeth.  Using the query below:
+
+```sparql 
+prefix foaf:  <http://xmlns.com/foaf/0.1/>
+prefix rel: <http://purl.org/vocab/relationship/>
+
+select ?x (count(distinct ?child) as ?children)
+from <tag:stardog:api:context:all>
+where {
+	?x foaf:name ?y;
+        rel:parentOf ?child
+    filter regex(?y, "^B[áa]eth$", "i")
+}
+group by ?x
+order by desc(?children)
+```
+
+The above query (with reasoning enabled) returns 15 results as shown below:
+
+| x                                                                              | children |
+| http://example.com/Rawl_B502/clann_aingeda.trig#Báeth                          | 12       |
+| http://example.com/Rawl_B502/genelach_úa_m-bairrche.trig#Bóeth                 |  2        |
+| http://example.com/Rawl_B502/genelach_benntraige.trig#Báeth                    |   1       |
+| http://example.com/LL/de_genelach_dail_messi_corbb.trig#Baeth-a245b020         |   1       |
+| http://example.com/LL/genelach_h_n-enechglais.trig#Baeth-50d79733              |   1       |
+| http://example.com/Rawl_B502/genelach_úa_m-bairrche.trig#Báeth                 |   1       |
+| http://example.com/Rawl_B502/item_úi_máeli_rubae_la_laigniu.trig#Báeth         |   1       |
+| http://example.com/Rawl_B502/genelach_ciannachta.trig#Báeth                    |   1       |
+| http://example.com/LL/genelach_h_mugroin_i_m-maig_liphi.trig#Baeth             |   1       |
+| http://example.com/Rawl_B502/genelach_ceníuil_báeth.trig#Báeth                 |   1       |
+| http://example.com/LL/forthart_fea.trig#Baeth                                  |   1       |
+| http://example.com/Rawl_B502/úib_luchta.trig#Báeth                             |   1       |
+| http://example.com/Rawl_B502/do_primforslointib_Lagen_inso.trig#Báeth-604282d0 |   1       |
+| http://example.com/Rawl_B502/genelach_úa_rossa_i_úa_n_dícolla.trig#Báeth       |   1       |
+| http://example.com/LL/eoganachta_casil.trig#Buith                              |   1       |
+
+We discover that
+`http://example.com/Rawl_B502/clann_aingeda.trig#Báeth` has 12
+children that are directly related to that URL and who is well suited
```

Comment:
> that are directly related to that URL

For those coming to this from a textual/historical point of view, perhaps clarify that this means they are all described as his children in one particular text in one particular manuscript (for which, see the URL).